diff -dur gens-2.15.2/src/gens/gens_core/cpu/sh2/sh2a.asm gens-2.15.2-execstack/src/gens/gens_core/cpu/sh2/sh2a.asm
--- gens-2.15.2/src/gens/gens_core/cpu/sh2/sh2a.asm	2005-03-08 22:02:54.000000000 +0100
+++ gens-2.15.2-execstack/src/gens/gens_core/cpu/sh2/sh2a.asm	2008-08-04 19:50:28.000000000 +0200
@@ -6129,3 +6129,8 @@
 
 %include "sh2_io.inc"
 
+
+%ifidn __OUTPUT_FORMAT__,elf
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+
diff -dur gens-2.15.2/src/gens/gens_core/cpu/z80/z80.asm gens-2.15.2-execstack/src/gens/gens_core/cpu/z80/z80.asm
--- gens-2.15.2/src/gens/gens_core/cpu/z80/z80.asm	2005-03-08 22:02:39.000000000 +0100
+++ gens-2.15.2-execstack/src/gens/gens_core/cpu/z80/z80.asm	2008-08-04 19:35:31.000000000 +0200
@@ -6234,4 +6234,7 @@
 		dw 08392h, 08793h, 08394h, 08795h, 08796h, 08397h, 08B98h, 08F99h
 
 
+%ifidn __OUTPUT_FORMAT__,elf
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
 
diff -dur gens-2.15.2/src/gens/gens_core/gfx/blit.asm gens-2.15.2-execstack/src/gens/gens_core/gfx/blit.asm
--- gens-2.15.2/src/gens/gens_core/gfx/blit.asm	2005-03-08 22:02:52.000000000 +0100
+++ gens-2.15.2-execstack/src/gens/gens_core/gfx/blit.asm	2008-08-04 19:44:32.000000000 +0200
@@ -2051,3 +2051,8 @@
 		pop ebp
 		emms
 		ret
+
+%ifidn __OUTPUT_FORMAT__,elf
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+
diff -dur gens-2.15.2/src/gens/gens_core/gfx/gfx_cd.asm gens-2.15.2-execstack/src/gens/gens_core/gfx/gfx_cd.asm
--- gens-2.15.2/src/gens/gens_core/gfx/gfx_cd.asm	2005-03-08 22:02:51.000000000 +0100
+++ gens-2.15.2-execstack/src/gens/gens_core/gfx/gfx_cd.asm	2008-08-04 19:50:51.000000000 +0200
@@ -973,4 +973,7 @@
 		pop ebx
 		ret
 
+%ifidn __OUTPUT_FORMAT__,elf
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
 
diff -dur gens-2.15.2/src/gens/gens_core/gfx/hq2x16.asm gens-2.15.2-execstack/src/gens/gens_core/gfx/hq2x16.asm
--- gens-2.15.2/src/gens/gens_core/gfx/hq2x16.asm	2005-03-08 22:02:52.000000000 +0100
+++ gens-2.15.2-execstack/src/gens/gens_core/gfx/hq2x16.asm	2008-08-04 19:45:02.000000000 +0200
@@ -1955,3 +1955,7 @@
     dd ..@cross8, ..@flag0,  ..@flag0,  ..@flag0,
     dd ..@flag0,  ..@flag0,  ..@flag0,  ..@flag0
 
+%ifidn __OUTPUT_FORMAT__,elf
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+
diff -dur gens-2.15.2/src/gens/gens_core/io/io.asm gens-2.15.2-execstack/src/gens/gens_core/io/io.asm
--- gens-2.15.2/src/gens/gens_core/io/io.asm	2005-03-08 22:03:20.000000000 +0100
+++ gens-2.15.2-execstack/src/gens/gens_core/io/io.asm	2008-08-04 19:43:46.000000000 +0200
@@ -1435,3 +1435,8 @@
 		pop ebx
 		pop eax
 		ret
+
+%ifidn __OUTPUT_FORMAT__,elf
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+
diff -dur gens-2.15.2/src/gens/gens_core/mem/mem_m68k.asm gens-2.15.2-execstack/src/gens/gens_core/mem/mem_m68k.asm
--- gens-2.15.2/src/gens/gens_core/mem/mem_m68k.asm	2005-03-08 22:02:39.000000000 +0100
+++ gens-2.15.2-execstack/src/gens/gens_core/mem/mem_m68k.asm	2008-08-04 19:45:39.000000000 +0200
@@ -1942,3 +1942,7 @@
 
 %include "mem_m68k_32x.inc"
 
+%ifidn __OUTPUT_FORMAT__,elf
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+
diff -dur gens-2.15.2/src/gens/gens_core/mem/mem_s68k.asm gens-2.15.2-execstack/src/gens/gens_core/mem/mem_s68k.asm
--- gens-2.15.2/src/gens/gens_core/mem/mem_s68k.asm	2005-03-08 22:02:39.000000000 +0100
+++ gens-2.15.2-execstack/src/gens/gens_core/mem/mem_s68k.asm	2008-08-04 19:45:59.000000000 +0200
@@ -3626,3 +3626,8 @@
 		pop edi
 		pop ecx
 		ret
+
+%ifidn __OUTPUT_FORMAT__,elf
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+
diff -dur gens-2.15.2/src/gens/gens_core/mem/mem_sh2.asm gens-2.15.2-execstack/src/gens/gens_core/mem/mem_sh2.asm
--- gens-2.15.2/src/gens/gens_core/mem/mem_sh2.asm	2005-03-08 22:02:39.000000000 +0100
+++ gens-2.15.2-execstack/src/gens/gens_core/mem/mem_sh2.asm	2008-08-04 19:46:41.000000000 +0200
@@ -2194,3 +2194,8 @@
 		pop edi
 		mov [_32X_VDP + vx.AF_St], edx
 		ret
+
+%ifidn __OUTPUT_FORMAT__,elf
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+
diff -dur gens-2.15.2/src/gens/gens_core/mem/mem_z80.asm gens-2.15.2-execstack/src/gens/gens_core/mem/mem_z80.asm
--- gens-2.15.2/src/gens/gens_core/mem/mem_z80.asm	2005-03-08 22:02:39.000000000 +0100
+++ gens-2.15.2-execstack/src/gens/gens_core/mem/mem_z80.asm	2008-08-04 19:46:19.000000000 +0200
@@ -499,4 +499,9 @@
 		push ecx
 		call M68K_WB
 		add esp, 16
-		ret
\ No newline at end of file
+		ret
+
+%ifidn __OUTPUT_FORMAT__,elf
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+
diff -dur gens-2.15.2/src/gens/gens_core/misc/misc.asm gens-2.15.2-execstack/src/gens/gens_core/misc/misc.asm
--- gens-2.15.2/src/gens/gens_core/misc/misc.asm	2005-03-08 22:03:18.000000000 +0100
+++ gens-2.15.2-execstack/src/gens/gens_core/misc/misc.asm	2008-08-04 19:36:04.000000000 +0200
@@ -1258,3 +1258,8 @@
 		pop ecx
 		pop ebx
 		ret
+
+%ifidn __OUTPUT_FORMAT__,elf
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+
diff -dur gens-2.15.2/src/gens/gens_core/sound/pwm.asm gens-2.15.2-execstack/src/gens/gens_core/sound/pwm.asm
--- gens-2.15.2/src/gens/gens_core/sound/pwm.asm	2005-03-08 22:02:37.000000000 +0100
+++ gens-2.15.2-execstack/src/gens/gens_core/sound/pwm.asm	2008-08-04 19:47:06.000000000 +0200
@@ -491,4 +491,9 @@
 	.End
 		pop esi
 		pop ebx
-		ret
\ No newline at end of file
+		ret
+
+%ifidn __OUTPUT_FORMAT__,elf
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+
diff -dur gens-2.15.2/src/gens/gens_core/vdp/vdp_32x.asm gens-2.15.2-execstack/src/gens/gens_core/vdp/vdp_32x.asm
--- gens-2.15.2/src/gens/gens_core/vdp/vdp_32x.asm	2005-03-08 22:02:40.000000000 +0100
+++ gens-2.15.2-execstack/src/gens/gens_core/vdp/vdp_32x.asm	2008-08-04 19:47:29.000000000 +0200
@@ -174,4 +174,9 @@
 			jnz short .Palette_Loop 
 
 		popad
-		ret
\ No newline at end of file
+		ret
+
+%ifidn __OUTPUT_FORMAT__,elf
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+
diff -dur gens-2.15.2/src/gens/gens_core/vdp/vdp_io.asm gens-2.15.2-execstack/src/gens/gens_core/vdp/vdp_io.asm
--- gens-2.15.2/src/gens/gens_core/vdp/vdp_io.asm	2005-03-08 22:02:40.000000000 +0100
+++ gens-2.15.2-execstack/src/gens/gens_core/vdp/vdp_io.asm	2008-08-04 19:47:53.000000000 +0200
@@ -1689,4 +1689,9 @@
 
 	.No_H_Int
 		and byte [_main68k_context + 35 * 4], 0xF0
-		ret
\ No newline at end of file
+		ret
+
+%ifidn __OUTPUT_FORMAT__,elf
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+
diff -dur gens-2.15.2/src/gens/gens_core/vdp/vdp_rend.asm gens-2.15.2-execstack/src/gens/gens_core/vdp/vdp_rend.asm
--- gens-2.15.2/src/gens/gens_core/vdp/vdp_rend.asm	2005-03-08 22:02:40.000000000 +0100
+++ gens-2.15.2-execstack/src/gens/gens_core/vdp/vdp_rend.asm	2008-08-04 19:48:12.000000000 +0200
@@ -2253,3 +2253,8 @@
 
 		popad
 		ret
+
+%ifidn __OUTPUT_FORMAT__,elf
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+
diff -dur gens-2.15.2/src/starscream/main68k/star.c gens-2.15.2-execstack/src/starscream/main68k/star.c
--- gens-2.15.2/src/starscream/main68k/star.c	2005-03-08 22:03:12.000000000 +0100
+++ gens-2.15.2-execstack/src/starscream/main68k/star.c	2008-08-04 19:29:23.000000000 +0200
@@ -2355,6 +2355,10 @@
 */
 static void suffixes(void) {
 	emit("end\n");
+	emit("\n");
+	emit("%%ifidn __OUTPUT_FORMAT__,elf\n");
+	emit("section .note.GNU-stack noalloc noexec nowrite progbits\n");
+	emit("%%endif\n");
 }
 
 /****************************************************************************
diff -dur gens-2.15.2/src/starscream/sub68k/star.c gens-2.15.2-execstack/src/starscream/sub68k/star.c
--- gens-2.15.2/src/starscream/sub68k/star.c	2005-03-08 22:03:08.000000000 +0100
+++ gens-2.15.2-execstack/src/starscream/sub68k/star.c	2008-08-04 19:42:40.000000000 +0200
@@ -2264,6 +2264,10 @@
 */
 static void suffixes(void) {
 	emit("end\n");
+        emit("\n");
+        emit("%%ifidn __OUTPUT_FORMAT__,elf\n");
+        emit("section .note.GNU-stack noalloc noexec nowrite progbits\n");
+        emit("%%endif\n");
 }
 
 /****************************************************************************
